source/puppet/4.10/lang_data_number.md

```diff
@@ -57,49 +56,63 @@ $product = 8 * 3e5  # evaluates to 240000.0
 
 Floats can't be expressed in octal or hex.
 
 ``` puppet
-# octal
-$value = 0777   # evaluates to decimal 511
-$value = 0789   # Error, invalid octal
-$value = 0777.3 # Error, invalid octal
-
-# hexadecimal
-$value = 0x777 # evaluates to decimal 1911
-$value = 0xdef # evaluates to decimal 3567
-$value = 0Xdef # same as above
-$value = 0xDEF # same as above
-$value = 0xLSD # Error, invalid hex
+# Octal
+$value = 0777    # evaluates to decimal 511
+$value = 0789    # Error, invalid octal
+$value = 0777.3  # Error, invalid octal
+
+# Hexadecimal
+$value = 0x777  # evaluates to decimal 1911
+$value = 0xdef  # evaluates to decimal 3567
+$value = 0Xdef  # same as above
+$value = 0xDEF  # same as above
+$value = 0xLSD  # Error, invalid hex
 ```
 
 ## Converting numbers to strings
 
 Numbers are automatically converted to strings when interpolated into a string. The automatic conversion uses decimal (base 10) notation.
 
-If you need to convert numbers to non-decimal string representations, you can use [the `sprintf` function.](./function.html#sprintf)
+To convert numbers to non-decimal string representations, you can use the [`sprintf` function](./function.html#sprintf), or you can create a new string object with the `new()` function and [specify its format](./function.html#conversion-to-string).
 
 ## Converting strings to numbers
 
-The [arithmetic operators][arithmetic] will automatically convert strings to numbers.
+[Arithmetic operators][arithmetic] automatically convert strings to numbers.
 
 In all other contexts (resource attributes, function arguments, etc.), Puppet _won't_ automatically convert strings to numbers, but you can:
 
-* Add 0 to manually convert a string to a number. (For example, `$mystring = "85"; $mynum = 0 + $mystring`.)
-* Use [the `scanf` function](./function.html#scanf) to manually extract numbers from strings. This function can also account for surrounding non-numerical text.
+-   Create a new `Integer` or `Float` object using the string as a parameter, such as `$mynum = Integer($mystring)`. This invokes the type's `new()` function, which can convert objects of other types --- including Strings --- into [Integers](./function.html#conversion-to-integer) and [Floats](./function.html#conversion-to-float).
+
+    The default radix of this conversion is base 10, but you can specify others.
+-   Use [the `scanf` function](./function.html#scanf) to manually extract numbers from strings. This function can also account for surrounding non-numerical text.
+
+> **Deprecation note:** Puppet 4.10.2 deprecated numeric coercion, which silently and automatically converted strings into numbers by adding a number to a string (for example, `$mynum = '10' + 0` would result in `$mynum` being assigned the number 10.) This now logs a Warning, or generates an Error if `--strict` is enabled.
+>
+> To ensure that your code works with future versions of Puppet, create new instances of numeric objects to convert String values to numeric types.
```

Comment:
Or - (obviously), avoid having numbers in string form in the first place... then there is no need to convert anything.